### PR TITLE
Media Library: Debounce search requests

### LINKED ISSUE #3090: Media Library: search box triggers requests for every keystroke
## Bug Description

The media library search box triggers requests for every keystroke when typing rapidly.

This is a waste and looks bad.

## Additional Context

- Plugin Version:
- Operating System:
- Browser:

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

### PR DESCRIPTION
**Pull request by diegovar**
## Summary

Adds a debouncer to the search input, so that we don't trigger a search for every. single. keystroke. when typing rapidly.

Before: https://share.getcloudapp.com/xQuDQJmQ

After: https://share.getcloudapp.com/JruL5ZZZ

## Relevant Technical Choices

using use-debouncer.

## User-facing changes

Typing in the search bar now waits a bit before triggering the search request.

## Testing Instructions

Ensure that search still works with the added timing change.

Fixes #3090

### CODE REVIEW

File: assets/src/edit-story/app/media/utils/getResourceFromMedia3p.js [hunk 1]
```diff
@@ -66,41 +66,34 @@ import createResource from './createResource';
  */
 function getUrls(m) {
   if (m.type.toLowerCase() === 'image') {
```
Comment (joannag6) [inline assets/src/edit-story/app/media/utils/getResourceFromMedia3p.js]: The changes in this file seem unrelated to this PR

**Review by joannag6 — APPROVED** [review]
LGTM

File: assets/src/edit-story/components/library/panes/media/mediaPane.js [hunk 2]
```diff
@@ -196,12 +200,20 @@ function MediaPane(props) {
   });
 
   /**
-   * Handle search term changes.
+   * Effectively performs a search, triggered at most every 500ms.
+   */
+  const [changeSearchTermDebounced] = useDebouncedCallback(() => {
```
Comment (swissspidy) [inline assets/src/edit-story/components/library/panes/media/mediaPane.js]: Why not just make `onSearch` debounced instead of introducing state?

Comment (diegovar) [reply assets/src/edit-story/components/library/panes/media/mediaPane.js]: SearchInput is a controlled component (its displayed value is the "searchTerm" which triggers a search), so unless you decouple the current value displayed in the input with the value that is used to trigger the searches, it doesn't work in my testing.  If you debounce onSearch, what happens is that a keystroke takes 500ms to register and appear in the input, which is not what we want.

File: assets/src/edit-story/app/media/utils/getResourceFromMedia3p.js (hunk 1 above)
Comment (diegovar) [reply assets/src/edit-story/app/media/utils/getResourceFromMedia3p.js]: Oops sorry, ended up in the wrong PR.

**Review by joannag6 — APPROVED** [review]

**Pull request merged**